tests/api/test_remote_engine.py

```diff
@@ -102,3 +103,62 @@ def test_device_class_target(self):
         target = "X8"
         engine = RemoteEngine(target)
         assert engine.target == engine.DEFAULT_TARGETS[target]
+
```

Comment:
Would it make sense to have some of these in a separate `class TestRemoteEngineIntegration`?

Reply:
I think `test_compilation` definitely! The others are not testing integration due to mocking, and are solely testing internal engine logic, so are probably fine to leave as is.

Will move `test_compilation`, and then merge.